backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/request/RequestRestController.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * #%L
+ * de.metas.business.rest-api-impl
+ * %%
+ * Copyright (C) 2025 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.rest_api.request;
+
+import de.metas.Profiles;
+import de.metas.bpartner.effective.BPartnerEffectiveBL;
+import de.metas.externalreference.rest.v2.ExternalReferenceRestControllerService;
+import de.metas.externalsystem.ExternalSystemRepository;
+import de.metas.logging.LogManager;
+import de.metas.rest_api.utils.JsonErrors;
+import de.metas.rest_api.v2.bpartner.BpartnerRestController;
+import de.metas.rest_api.v2.bpartner.bpartnercomposite.JsonRetrieverService;
+import de.metas.rest_api.v2.ordercandidates.impl.MasterdataProvider;
```

Comment:
IMHO you shall introduce your MasterdataProvider, maybe call it RequestMasterdataProvider (new class).
In implementation, if needed u can use`de.metas.rest_api.v2.ordercandidates.impl.MasterdataProvider`, even though i would prefer to not but instead just extract refactor parts of it.


then all methods like resolveProductIdOrNull, resolveUserIdOrNull, etc shall go into RequestMasterdataProvider (not static!).